pkg/globalnet/controllers/egress_pod_watcher.go

```diff
@@ -95,8 +96,8 @@ func (w *egressPodWatcher) onCreateOrUpdate(obj runtime.Object, _ int) bool {
 
 	logger.V(log.DEBUG).Infof("Pod %q with IP %s created/updated", key, pod.Status.PodIP)
 
-	if err := w.namedIPSet.AddEntry(pod.Status.PodIP, true); err != nil {
-		logger.Errorf(err, "Error adding pod IP %q to IP set %q", pod.Status.PodIP, w.ipSetName)
+	if err := w.pfIface.AddEgressPodIP(w.gnHandle, pod.Status.PodIP); err != nil {
```

Comment:
Suppose we have a use-case where we want all the pods from a certain namespace to use a specific egressIP, we create a namedIPSet which has a unique name created using the key (i.e., namespace/name) and and store the PodIPs that are part of that namespace. It is not clear if we are doing the same in the new code. Can you explain how we are handling it? Is gnHandle supposed to be a unique name derived based on the key?